dotnet/src/Microsoft.AutoGen/AgentChat/GroupChat/OutputCollectorAgent.cs

```diff
@@ -43,7 +43,12 @@ private void RunSynchronized(Action<SinkFrame> frameAction)
             frameAction(this.receivingSinkFrame);
         }
 
-        semapohre.Release();
+        // TODO: Replace the Semaphore with a TaskSource approach
+        try
+        {
+            semapohre.Release();
+        }
+        catch (SemaphoreFullException) { }
```

Comment:
should we at least emit a log?

Reply:
No, there is no actual error here. This is just a bit of an abuse of Semaphore semantics.